### PR TITLE
Query page (probably) finished. 

### PR DESCRIPTION
**Pull request by jtr-usaf**
Query page finished, initial query_results page as well. 
Query page pictured below.
image

### CODE REVIEW

File: src/app/__init__.py [hunk 1]
```diff
@@ -23,7 +24,9 @@
 
 @app.context_processor
 def inject_oidc():
```
Comment (DL-USAF) [inline src/app/__init__.py:26]: can we call this something else since we now have more than just oidc here?

File: src/app/routes.py [hunk 2]
```diff
@@ -1,5 +1,12 @@
 from flask import render_template, Blueprint, redirect, url_for, request
 from app import oidc
+from click.testing import CliRunner
```
Comment (DL-USAF) [inline src/app/routes.py:3]: is click a dependency in the toml?

File: src/app/routes.py [hunk 3]
```diff
@@ -21,27 +28,43 @@ def dashboard():
...
+
+    return render_template('query.html', auths=auths)
+
 
+@main.route('/query_results', methods=['GET', 'POST'])
+@oidc.require_login
+def query_results():
     if request.method == 'POST':
-        query_name = request.form.get('query_name')
+        query_name = request.form.get('query_name') or uuid4()
```
Comment (jtr-usaf) [inline src/app/routes.py:45]:
```suggestion
        query_name = request.form.get('query_name', uuid4())
```

File: src/app/routes.py [hunk 4]
```diff
@@ -21,27 +28,43 @@ def dashboard():
...
-                               decode_raw_data=decode_raw_data)
+        decode_raw_data = bool(request.form.get('decode_raw_data'))
+
+        output_location = f'/tmp/{query_name}/results.json'
+        cmd = [
+            'query',
+            '-c', 'PLACEHOLDER', '-k', 'PLACEHOLDER',
+            '-q', query_text,
+            '--query-name', f'{query_name}',
+            '--auths', ",".join(selected_auths),
```
Comment (jtr-usaf) [inline src/app/routes.py:57]: Should add a way to guarantee they selected at least one auth

File: src/mock/datawave_cli/main.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        <li><strong>Output Location:</strong> {output}</li>
+        <li><strong>Decode Raw Data:</strong> {decode_raw}</li>
+    </ul>
+    """
+
+
+@main.command
+@common_options
+def authorization(**kwargs):
+    res = {'proxied_users': [{'auths': ['FOO', 'BAR', 'PUBLIC', 'PRIVATE']}]}
```
Comment (DL-USAF) [inline src/mock/datawave_cli/main.py:48]: add a section for role... did you base this off what we normally get? or should I get an example?

File: src/app/routes.py (hunk 3 above)
Comment (jtr-usaf) [reply src/app/routes.py:45]: Just checked. This does not work because it does retreive something. It's just blank.

File: src/app/routes.py (hunk 4 above)
Comment (jtr-usaf) [reply src/app/routes.py:57]: image

**Review by DL-USAF — APPROVED** [review]

**Pull request merged**